lib/resources/elasticsearch.rb

```diff
@@ -0,0 +1,140 @@
+# encoding: utf-8
+# copyright: 2017
+# author: Rony Xavier, rx294@nyu.edu
+# author: Aaron Lippold, lippold@gmail.com
+# license: All rights reserved
+
+require 'utils/filter'
+
+module Inspec::Resources
+  class Elasticsearch < Inspec.resource(1)
+    name 'elasticsearch'
+    desc 'See Description below'
+    # Use the Elasticsearch InSpec audit resource to test the Elasticsearch
+    # configuration settings. The data is extract from the live elastic search
+    # cluster using the curl command `curl localhost:9200/_nodes/`. The resource
+    # makes uses of FilterTable and its functionality can be used
+
+    example 'See example below'
+    # elasticsearch.where{ cluster_name == 'elasticsearch' }.version.each do |node_version|
+    #   describe node_version do
+    #     it {should eq '5.4.1'}
+    #   end
+    # end
+
+    attr_reader :params, :content
+
+    def initialize(es_ip = 'localhost', es_port = 9200)
+      @es_ip   = es_ip
+      @es_port = es_port
+
+      # skip if connection refused
+      @content = inspec.command("curl http://#{es_ip}:#{es_port}/_nodes/").stdout
```

Comment:
The `connection refused` output will be output on stderr, not stdout. You're likely going to need to do something like this:

```ruby
command = inspec.command("curl http://#{es_ip}:#{es_port}/_nodes/")
if command.stderr =~ /Connection refused/
  return skip_resource ....
end

@content = JSON.parse(command.stdout)
```

You will also need to guard against JSON.parse throwing an exception or you'll cause InSpec to completely stop testing if the result is not JSON.

Please refactor the fetching and parsing of data into a helper method that can be tested rather than having it directly in the `initialize` method. Thanks!